runtime/common/src/mock.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Mocking utilities for testing.
+
+use std::{cell::RefCell, collections::HashMap};
+use parity_scale_codec::{Encode, Decode};
+use sp_runtime::traits::SaturatedConversion;
+use frame_support::dispatch::DispatchResult;
+use primitives::v1::{HeadData, ValidationCode, Id as ParaId};
+use crate::traits::Registrar;
+
+thread_local! {
+	static OPERATIONS: RefCell<Vec<(ParaId, u32, bool)>> = RefCell::new(Vec::new());
+	static PARACHAINS: RefCell<Vec<ParaId>> = RefCell::new(Vec::new());
+	static PARATHREADS: RefCell<Vec<ParaId>> = RefCell::new(Vec::new());
+	static MANAGERS: RefCell<HashMap<ParaId, Vec<u8>>> = RefCell::new(HashMap::new());
+}
```

Comment:
parameter_types already supports this as well: 

```
parameter_types! {
    static Operations: Vec<(ParaId, u32, bool)> = Vec::new();
}

will generate both the `thead_local` stuff with SCREAM_CASE, and a struct named `Operations` that implements `Get`, and you can easily do `Operations::set(value)` to set the value of the `thread_local` value.

Reply:
See example here: https://crates.parity.io/frame_support/macro.parameter_types.html